src/views/ViewAllExplorer.tsx

```diff
@@ -39,27 +37,33 @@ function ViewAllExplorer({
   return (
     <Animated.View style={{ opacity: fadeAnim }}>
       <>
-        <NavHeader title="Connect your Wallet" onBackPress={onBackPress} />
-        {isLoading || !currentWCURI ? (
+        <NavHeader
+          title="Connect your Wallet"
+          onBackPress={RouterCtrl.goBack}
+        />
+        {loading ? (
           <ActivityIndicator
             style={styles.loader}
             color={isDarkMode ? LightTheme.accent : DarkTheme.accent}
           />
         ) : (
           <FlatList
-            data={explorerData || []}
+            data={wallets || []}
             style={styles.list}
             contentContainerStyle={styles.listContentContainer}
             indicatorStyle={isDarkMode ? 'white' : 'black'}
             showsVerticalScrollIndicator
             numColumns={4}
-            getItemLayout={(_, index) => ({
+            getItemLayout={(__, index) => ({
```

Comment:
`__` looks weird, maybe we can have a more descriptive unused variable i.e `_something`